app/src/format.py

```diff
@@ -369,13 +376,24 @@ def reify_citations(
 
 # FIXME: Refactor to reduce code replication with replace_citation_ids()
 def _add_citation_links(
-    response: str, remapped_citations: dict[str, Subsection], config: FormattingConfig
+    response: str,
+    remapped_citations: dict[str, Subsection],
+    config: FormattingConfig,
+    map_of_accordion_ids: dict,
 ) -> str:
     global _footnote_id
     _footnote_id += 1
     footnote_list = []
 
     # Replace (citation-<index>) with the appropriate citation
+
+    def find_accordion_id(citation_num: str) -> str | None:
+        if map_of_accordion_ids:
+            for key, value in map_of_accordion_ids.items():
+                if citation_num in value:
+                    return key
+        return None
```

Comment:
I think you should flip the key/value of map_of_accordion_ids: each key should be a citation number, and the value under that key should be the accordion ID. That makes this function
```
def find_accordion_id(citation_num: str) -> str | None:
    return map_of_accordion_ids[citation_num] if citation_num in map_of_accordion_ids else None
```
 which is small enough that you could probably do it directly on line 408